### PR TITLE
handled a missing suspense fiber when suspense is filtered on the profiler

### PR DESCRIPTION
**Pull request by IDrissAitHafid**
fixes [#18256](https://github.com/facebook/react/issues/18256#issue-578077772) and fixes [#19633](https://github.com/facebook/react/issues/19633#issue-680756123)

If suspense is filtered in the profiler and started profiling, when the suspense fallback unmounts, the primary fiber tries to mount to the suspense fiber above it, but because it is filtered the profiler throws those errors thus the missing fiber in this [#19633](https://github.com/facebook/react/issues/19633#issue-680756123) and the undefined in this [#18256](https://github.com/facebook/react/issues/18256#issue-578077772). 
So instead of that, when suspense is filtered I re-parent it to the parent fiber of the suspense fiber.

That said, this problem also generates this issue [#19547](https://github.com/facebook/react/issues/19547#issue-674411480) and this one [#18924](https://github.com/facebook/react/issues/18924#issue-618218428). Depending on the place of the suspense fiber in the tree relatively to the rendered elements (is it rendered alone, or rendered with siblings, or only its children are rendered...) a different error is thrown.
In the example below suspense is rendered along with a `<h1>Hello </h1>`  sibling and a div parent.

**Before:**
before the fix

**After:**
after the fix

### CODE REVIEW

**Comment by IDrissAitHafid** [comment]
Here's the [code sandbox app](https://codesandbox.io/s/missing-suspense-error-repro-c2zjy?file=/src/App.js) of the repro.

**Comment by IDrissAitHafid** [comment]
@bvaughn can you please review this PR 🙏 😅

**Comment by bvaughn** [comment]
Nice fix!

I literally just sat down to look into these issues 10 minutes ago, so I appreciate the ping in this case. I didn't notice you'd opened this.

**Comment by IDrissAitHafid** [comment]
That's what I thought 😄 
Thanks!

File: packages/react-devtools-shared/src/backend/renderer.js [hunk 1]
```diff
@@ -1572,7 +1572,7 @@ export function attach(
       if (nextPrimaryChildSet !== null) {
         mountFiberRecursively(
           nextPrimaryChildSet,
-          nextFiber,
+          shouldIncludeInTree ? nextFiber : parentFiber,
```
Comment (bvaughn) [inline packages/react-devtools-shared/src/backend/renderer.js:1575]: Huh. Looks like we handled this correctly in 5 of the 6 places, but missed this one. Great find! :grin:

**Review by bvaughn — APPROVED** [review]

**Comment by bvaughn** [comment]
Am I understanding correctly that this PR fixes #18256, #19633, #19547, and #18924?

**Comment by IDrissAitHafid** [comment]
I think I didn't explain well what it is fixed 😅
this PR fixed exactly this one [#18256](https://github.com/facebook/react/issues/18256#issue-578077772), and this one [#19633](https://github.com/facebook/react/issues/19633#issue-680756123) is always thrown whenever a fiber is missing.

 [#18924](https://github.com/facebook/react/issues/18924) is not fixed, when I have time I'll share how to reproduce it and if you want I'll work on it and see if I can find something 😅, this bug  [#18924](https://github.com/facebook/react/issues/18924) also generates this one [#19633](https://github.com/facebook/react/issues/19633#issue-680756123). That's why I couldn't tell if [#19633](https://github.com/facebook/react/issues/19633#issue-680756123)  is fixed.

File: packages/react-devtools-shared/src/backend/renderer.js (hunk 1 above)
Comment (IDrissAitHafid) [reply packages/react-devtools-shared/src/backend/renderer.js:1575]: Thank you!

**Comment by bvaughn** [comment]
Are you confident that #18924 is *not fixed*? Or are you not confident that it *is* fixed?

**Comment by bvaughn** [comment]
Thanks for clarifying! I will reopen for now then :)

**Pull request merged**